rotkehlchen/chain/evm/decoding/curve/curve_cache.py

```diff
@@ -390,59 +410,101 @@ def _query_curve_data_from_chain(
             method_name='get_address',
             arguments=[7],
         ))
-    except DeserializationError as e:
-        log.error(f'Curve address provider returned an invalid address for metaregistry. {e}')
-        return []
+    except (RemoteError, DeserializationError) as e:
+        log.error(
+            f'Failed to retrieve metaregistry address from the Curve '
+            f'address provider on {evm_inquirer.chain_name} due to {e!s}',
+        )
+        return
 
     metaregistry = EvmContract(
         address=metaregistry_address,
         abi=evm_inquirer.contracts.abi('CURVE_METAREGISTRY'),  # type: ignore[call-overload]  # for some reason mypy doesn't properly see argument type
         deployed_block=0,  # deployment_block is not used and the contract is dynamic
     )
-    pool_count = metaregistry.call(node_inquirer=evm_inquirer, method_name='pool_count')
-    new_pools: list[CurvePoolData] = []
+    try:
+        pool_count = metaregistry.call(node_inquirer=evm_inquirer, method_name='pool_count')
+    except RemoteError as e:
+        log.error(
+            'Failed to retrieve Curve pool count from the metaregistry '
+            f'on {evm_inquirer.chain_name} due to {e!s}',
+        )
+        return
+
+    if (existing_pool_count := len(existing_pools)) == pool_count:
+        return
+    if (new_pool_count := pool_count - existing_pool_count) > MAX_ONCHAIN_POOLS_QUERY:
+        pool_count = existing_pool_count + MAX_ONCHAIN_POOLS_QUERY
+        new_pool_count = MAX_ONCHAIN_POOLS_QUERY
+        log.error(
```

Comment:
This is not an error.

```suggestion
        log.info(
```